### PR TITLE
AB test framework

### PR DESCRIPTION
**Pull request by svillafe**
## Why are you doing this?

We want to be able to test features using A/B tests, in this PR we add a shared helper which handle the AB tests logic and an example test that set up different messages for a title on the landing page. The example test is turn off currently.
 
[**Trello Card**](https://trello.com/c/diAPTYG7/557-a-b-testing-module)

## Screenshots
N/A

### CODE REVIEW

File: assets/helpers/abtest.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+const MVT_COOKIE: string = 'GU_mvt_id';
+const MVT_MAX: number = 1000000;
+
+
+// ----- Types ----- //
+
+type OtherWaysOfContributeVariants =
+  | 'control'
+  | 'notintest'
```
Comment (Ap0c) [inline assets/helpers/abtest.js]: 426236e

File: assets/helpers/abtest.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    const test = tests[testId];
+
+    if (!test.isActive) {
+      return;
+    }
+
+    if (testId in currentParticipation) {
+      participation[testId] = currentParticipation[testId];
+    } else if (userInTest(test.audience, mvtId)) {
+      const variantIndex = mvtId % test.variants.length;
```
Comment (davidfurey) [inline assets/helpers/abtest.js]: Maybe factor this out into a function "assign user to variant"

File: assets/helpers/abtest.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+type Test = {
+  variants: string[],
+  audience: Audience,
+  isActive: boolean,
+};
+
+
+// ----- Tests ----- //
+
+const tests: {
```
Comment (davidfurey) [inline assets/helpers/abtest.js]: I think this could be an array

Comment (Ap0c) [inline assets/helpers/abtest.js]: 1160858

File: assets/pages/bundles-landing/helpers/abtest.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+// ----- Functions ----- //
+
+const otherWaysOfContribute = (participation: Participation): string => {
+
+  const variant = participation.otherWaysOfContribute;
+
+  switch (variant) {
+    case 'control' : return 'other ways you can support us';
+    case 'variantA' : return 'other ways you can contribute';
+    case 'variantB' : return 'other ways you can give us money';
```
Comment (davidfurey) [inline assets/pages/bundles-landing/helpers/abtest.js]: This is a good point.  I think we will need to do some refactoring of the a/b test framework as we run the first few tests.  The aim should be to need the minimum amount of work for each a/b test.

Comment (svillafe) [inline assets/pages/bundles-landing/helpers/abtest.js]: Another approach for reducing the amount of code could be creating a shared presentation module and leave the ab-test.js module as simple as possible. This kind of shared component would be justified in the case of a test which impacts several pages. In the example test, we created a non-shareable presentation module for the a/b test which only impacts the bundle landing page..

**Pull request merged**